src/redux/store.ts

```diff
@@ -5,43 +5,41 @@ import {
   CombinedState
 } from 'redux'
 import logger from 'redux-logger'
-import rootReducer, { blankState } from './reducer'
+import rootReducer from './reducer'
 
-import { persistStore, persistReducer, createTransform } from 'redux-persist'
+import { persistStore, persistReducer, createMigrate } from 'redux-persist'
 import storage from 'redux-persist/lib/storage' // defaults to localStorage for web
 import { Information } from 'ustaxes/core/data'
 import { blankYearTaxesState, YearsTaxesState } from '.'
 import { Actions } from './actions'
 import { PersistPartial } from 'redux-persist/es/persistReducer'
 import { FSAction } from './fs/Actions'
 import fsReducer from './fs/FSReducer'
+import { migrateEachYear } from './migration'
 
-const baseTransform = createTransform(
-  // transform state on its way to being serialized and persisted.
-  (inboundState: Information) => {
-    return inboundState
-  },
-  // transform state being rehydrated
-  // Just ensure the state has all requisite root members
-  (outboundState: Information): Information => {
-    return {
-      ...blankState,
-      ...outboundState
-    }
-  },
-  { whitelist: ['information'] }
-)
-
-const persistConfig = {
-  key: 'root',
-  storage,
-  transforms: [baseTransform]
-}
+/**
+ * Each key of these migrations will only run if it is less than the current version.
+ * Even though this is a general purpose migration function that should handle most changes,
+ * a new key will need to be added here with each change to the state shape.
+ **/
+const migrate = createMigrate({
+  '0': (state) => {
+    // this type dance is unfortunate, but createMigrate isn't typed generically...
+    const migrated = migrateEachYear(state as unknown as YearsTaxesState)
+    return migrated as unknown as typeof state
```

Comment:
I bet you can get rid of this. I agree I don't like the type in `redux-persist` could probably be better, but it's just requiring that the parameter and return is a subtype of `PersistedState`. So you can enforce any union type of `PersistedState` in the migration functions, see comment in `migration.ts`.

Reply:
Found a much cleaner way by dropping their whole system for a plain async function with just our simple logic.